keyboards/chidori/keymaps/oled_sample/keymap.c

```diff
@@ -0,0 +1,224 @@
+      _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,            KC_F12,  _______, _______, _______, _______, _______,
+      _______, _______, _______, _______, _______, _______,           _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
+    ),
+
+  /* Adjust (Lower + Raise)
+   * ,-----------------------------------------.             ,-----------------------------------------.
+   * |      | Reset|      |      |      |      |             |      |Qwerty|Colemk|Dvorak|      | Ins  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | Caps |      |      |      |      | Mac  |             | Win  |  -   |   =  |Print |ScLock|Pause |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |      |      |      |      |      |             |      |      |      |      |      |      |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |      |      |      |      |      |             | KANA |      | Home |PageDn|PageUp| End  |
+   * `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_ADJUST] = LAYOUT(
+      _______, RESET,   _______, _______, _______, _______,           _______, QWERTY,  COLEMAK, DVORAK,  _______, KC_INS,
+      KC_CAPS, _______, _______, _______, _______, AG_NORM,           AG_SWAP, KC_MINS, KC_EQL,  KC_PSCR, KC_SLCK, KC_PAUS,
+      _______, _______, _______, _______, _______, _______,           _______, _______, _______, _______, _______, _______,
+      _______, _______, _______, _______, _______, _______,           _______, _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END
+    )
+};
+// clang-format on
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+        case COLEMAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            return false;
+        case DVORAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_DVORAK);
+            }
+            return false;
+    }
+    return true;
+}
+
+bool led_update_user(led_t led_state) {
+    board_set_led_by_index(0, LED_YELLOW, led_state.caps_lock);
+    board_set_led_by_index(1, LED_YELLOW, led_state.scroll_lock);
+
+    return false;
+}
+
+#ifdef OLED_DRIVER_ENABLE
+
+void oled_write_layer_state(void) {
+    oled_write_P(PSTR("Layer: "), false);
+    switch (get_highest_layer(layer_state)) {
```

Comment:
Just thinking about this, and looking into the layer code more, it may be better to just call: 
```suggestion
    switch (get_highest_layer(layer_state|default_layer_state)) {
```

This merges both bitmasks, so that only the one switch is needed, and no nested switch is needed.